src/tir/analysis/buffer_access_lca_detector.cc

```diff
@@ -107,6 +122,17 @@ class LCADetector : public StmtExprVisitor {
     ancestor_scopes_.pop_back();
   }
 
+  void VisitStmt_(const AttrStmtNode* op) final {
+    if (op->attr_key == attr::thread_extent) {
+      const auto* iter = op->node.as<IterVarNode>();
+      ICHECK_NOTNULL(iter);
+      if (StartsWith(iter->thread_tag, "blockIdx.")) {
```

Comment:
Instead of string matching, use ThreadScope here https://github.com/apache/tvm/blob/main/src/runtime/thread_storage_scope.h#L161

Reply:
Thanks for pointing it out!! Just updated :-)